docs/feature_grave_esc.md

```diff
@@ -4,14 +4,22 @@ If you're using a 60% keyboard, or any other layout with no F-row, you will have
 
 ## Usage
 
-Replace the `KC_GRAVE` key in your keymap (usually to the left of the `1` key) with `KC_GESC`. When pressed it will behave like `KC_ESC`, but with Shift or GUI held it will send `KC_GRAVE`.
+Replace the `KC_GRAVE` key in your keymap (usually to the left of the `1` key) with `KC_GESC`. Most of the time this key will output `KC_ESC` when pressed. However, when Shift or GUI are held down it will output `KC_GRV` instead.
+
+## What Your OS sees
+
+If Mary presses GESC on her keyboard, the OS will see an KC_ESC character. Now if Mary holds Shift down and presses GESC it will output `~`, or a shifted backtick. Now if she simply holds GUI/CMD/WIN, it will output a simple <code>&#96;</code> character.
```

Comment:
Disagree it's like you just inverted the examples. Bike shedding.

Reply:
It's very important to be *consistent* in documentation, particularly the representation of keycodes, their output, and key combinations. Not bikeshedding at all. (maybe a little nitpicky)

Reply:
I don't understand how I've "inverted the examples". All the instances of `GESC` I've replaced with `KC_GESC` because that's not a real keycode. The OS doesn't see `KC_ESC`, it sees the numerical code that `KC_ESC` is defined as. For "plain English" reasons all non-printable characters should be represented by their name - hence Escape.

Reply:
Also, the "simply" and "simple" in close proximity looked a bit odd grammatically/sentence structurally.

Reply:
fair.

Reply:
KC_ESC is actually a mappable ANSI code. 27. You can capture it in unix with Ctrl + V. I think we're splitting hairs at this point.

The point stands. Before these examples, this documentation was incomplete. Even a technical user had difficulty understanding how the feature worked without trying it themselves. That should be a huge red flag. QMK is everyone, not just the technically literate. Sometimes we as humans make simplifications to improve understanding. This document was misleading before, and this change is to rectify that. Please don't conflate technical accuracy with clear communication.

Reply:
> KC_ESC is actually a mappable ANSI code. 27.

The ANSI code for Escape doesn't seem to be referred to as "KC_ESC" anywhere. QMK doesn't send the ANSI code anyway, it sends the HID usage for "Keyboard ESCAPE" which is `0x29`. Why not just say "Escape"??? Much easier for newbies to understand!

> QMK is for everyone, not just the technically literate

Yes, which is why I don't like using `KC_*` everywhere unless referring to the keycode define specifically eg. "put this keycode in your keymap". Nobody calls their Escape key the "KC_ESC" key! My goal here is readability and clear communication, like you.

Reply:
Ironically it was like that before but skully didn't like it. There was a discussion on discord. Sorry you weren't there.

Reply:
@skullydazed, if you don't mind could you elaborate on this? I don't quite understand the idea here.

Reply:
> If Mary presses GESC on her keyboard

My comments revolved around ESC vs GESC in this part, and other similar usage. We were trying to resolve the question of whether users would be less confused by referring to the letters printed on the keycap they're likely pressing (ESC) vs referring to the underlying key as QMK sees it (GESC).

Right now it seems to me that we all agree conceptually on how to divide these up into categories, but we disagree about what each name and/or roll each category is assigned to.

Reply:
Okay then. Well, as I mentioned before, "GESC" is not a real keycode in QMK, and nor is "ESC". Referring to these keycodes without the `KC_` prefix introduces confusion because some keycodes have a different or no prefix. I assumed that @yanfali meant "KC_GESC", so I suggested changing it to that, and putting it in code tags. I think all keycode "literals" should be in code tags to show that they are things you can place in a keymap, and because they are in fact bits of code.

With keys specified like `A` or "Left Shift" the usage is different. This representation is for demonstrating the end result when you press the key. Pressing `A` in a text editor gives you the letter "a" on the screen (for simplicity only upper case is used - see pretty much every context menu shortcut ever).

Left Shift and Escape don't print anything, so we should simply use the name of the key. Since `KC_GESC` does multiple things depending on other things, the name Grave Escape should be used if we don't want to use the literal here. Alternatively, the wording could be changed to indicate that Mary has set her Escape key to `KC_GESC`, then pressed Escape.